src/managers/RoleManager.js

```diff
@@ -143,6 +144,53 @@ class RoleManager extends BaseManager {
       });
   }
 
+  /**
+   * Edits a role of the guild.
+   * @param {RoleResolvable} role The role to edit
+   * @param {RoleData} data The new data for the role
+   * @param {string} [reason] Reason for editing this role
+   * @returns {Promise<Role>}
+   * @example
+   * // Edit a role
+   * guild.roles.edit('222079219327434752', { name: 'buddies' })
+   *   .then(updated => console.log(`Edited role name to ${updated.name}`))
+   *   .catch(console.error);
+   */
+  async edit(role, data, reason) {
+    const id = this.resolveID(role);
+    if (!id) throw new TypeError('INVALID_TYPE', 'role', 'RoleResolvable');
+
+    if (typeof data.position === 'number') {
+      const updatedRoles = await setPosition(
+        this,
+        data.position,
+        false,
+        this.guild._sortedRoles(),
+        this.client.api.guilds(this.guild.id).roles,
+        reason,
+      );
+
+      this.client.actions.GuildRolesPositionUpdate.handle({
+        guild_id: this.guild.id,
+        roles: updatedRoles,
+      });
+    }
+
+    const _data = {
+      name: data.name,
+      color: typeof data.color === 'undefined' ? undefined : resolveColor(data.color),
+      hoist: data.hoist,
+      permissions: typeof data.permissions === 'undefined' ? undefined : new Permissions(data.permissions),
+      mentionable: data.mentionable,
+    };
+
+    const d = this.client.api.guilds[this.guild.id].roles[id].patch({ data: _data, reason });
```

Comment:
```suggestion
    const d = this.client.api.guilds(this.guild.id).roles(id).patch({ data: _data, reason });
```
Small nit but iirc it was decided `()` was preferred over `[]` at some point so this is for consistency.